data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java

```diff
@@ -603,6 +617,82 @@ public void testServerStartCertFileSuccess() throws IOException {
         }
     }
 
+    @Test
+    public void testServerStartCertFileMissing() {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
+            armeriaServerMock.when(Server::builder).thenReturn(serverBuilder);
+
+            when(sourceConfig.isSsl()).thenReturn(true);
+            when(sourceConfig.getSslCertificateFile()).thenReturn(null);
+            when(sourceConfig.getSslKeyFile()).thenReturn(null);
+            HTTPSourceUnderTest = new HTTPSource(sourceConfig, pluginMetrics, pluginFactory, pipelineDescription);
+            assertThrows(NullPointerException.class, () -> HTTPSourceUnderTest.start(testBuffer));
+        }
+    }
+
+    @Test
+    void testServerStartACMCertSuccess() throws IOException {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
```

Comment:
Please try to reduce the amount of code in the `try` block for the `mockStatic`. I suspect you need it only when calling `HTTPSourceUnderTest.start`.

```
HTTPSourceUnderTest = new HTTPSource(sourceConfig, pluginMetrics, pluginFactory, pipelineDescription);

Field field = HTTPSourceUnderTest.getClass().getDeclaredField("certificateProviderFactory");
field.setAccessible(true);
field.set(HTTPSourceUnderTest, certificateProviderFactory);

try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
   HTTPSourceUnderTest.start(testBuffer);
}

HTTPSourceUnderTest.stop();
```

Reply:
🫡